tests/test_from_file.py

```diff
@@ -0,0 +1,128 @@
+from argparse import ArgumentError
+from dataclasses import dataclass, field
+from json import dumps
+from pathlib import Path
+from typing import Optional
+
+from yaml import dump
+
+from pytest import raises, warns
+
+from pydargs import parse
+
+
+@dataclass
+class SubSubConfig:
+    q: str = "q"
+
+
+@dataclass
+class SubConfig:
+    a: int = 42
+    b: str = field(default="abc", metadata=dict(help="a string"))
+    s: SubSubConfig = field(default_factory=SubSubConfig)
+
+
+class TestParseFromFile:
+    @dataclass
+    class Config:
+        a: int = 5
+        b: str = field(default="something", metadata={"ignore_arg": True})
+        c: bool = field(default=False, metadata=dict(ignore_arg=False, as_flags=True))
+        s: SubConfig = field(default_factory=SubConfig)
+        z: str = "dummy"
+
+    def test_parse_no_file(self) -> None:
+        config = parse(self.Config, [], load_from_config=True)
```

Comment:
Should this log a warning? i.e. setting load_from_config to True but not providing the key pointing to the file?

Reply:
The `load_from_config` flag enables the argument, allowing the user to provide a config file. I don't think it should be required or in any case pushed to use it when enabled.